toolchain/check/handle_binding_pattern.cpp

```diff
@@ -278,28 +240,37 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
             });
       }
       context.node_stack().Push(node_id, param_pattern_id);
-
-      // TODO: Use the pattern insts to generate the pattern-match insts
-      // at the end of the full pattern, instead of eagerly generating them
-      // here.
       break;
     }
 
-    case Parse::NodeKind::LetIntroducer: {
+    case FullPatternStack::Kind::NameBindingDecl: {
       cast_type_id = context.AsCompleteType(cast_type_id, type_node, [&] {
```

Comment:
The old code above did `AsConcreteType` but this is `AsCompleteType` which is similar but different - is that intended? Can you teach me why?

Reply:
> That change is basically independent of the rest of the PR, so I can revert it if you like.

As noted elsewhere, I decided to revert it.